### ISSUE TITLE
Improve Native CLI size

### ISSUE THREAD

**Issue by DanielHabenicht** [kyubisation/angular-server-side-configuration#23]
I really love your solution for providing compatibility with angular `--aot` and better type support than stitching everything together with some custom scripts. (You may have read my comment [here](https://github.com/angular/angular-cli/issues/3855#issuecomment-461637797))

But I am a fan of small docker images. The native cli approach currently provided adds 40-50MB to a docker image. That's way to much for just file replacements. 

So the questions is: "How to reduce the file size?" 

The only option I can think about is rewriting the cli in a language that can be compiled to a native platform (with a much lighter footprint, e.g. Golang) and hooking it up to npm while also providing the option to copy the native files. 

What do you think about it? 

PS:
I've opened an issue here as I don't want to spam the angular-cli issue and also don't want to develop another solution for the same problem. I am highly motivated to implement a more lightweight solution and just wanted to hear your opinion on this topic.

**Comment by kyubisation** [comment]
Thank you and you have a fair point.
However, I'm not keen on changing the existing implementation.
What I could imagine is providing scripts in different languages/binaries for the `insert` case (because that is the only one, that I think would be used standalone).
I have however not previously used golang. From what I understand, the golang code could be compiled  to a wide range of different binaries (https://golang.org/doc/install/source#environment), so it would certainly make sense.
So basically while I don't want to change existing code, I'm open to ideas for `insert` alternatives.

**Comment by kyubisation** [comment]
I have released version 2.0.0. It includes documentation for using the native cli: https://www.npmjs.com/package/angular-server-side-configuration#native-cli

The final size of the executables is less than a MB.

Thank you very much for your help 👍
